sympy/printing/pycode.py

```diff
@@ -488,6 +490,16 @@ class NumPyPrinter(PythonCodePrinter):
     ))
     _kc = {k: 'numpy.'+v for k, v in _known_constants_math.items()}
 
+    def __init__(self, settings=None):
+        # XXX: Needed to avoid circular import
+        from sympy.codegen.rewriting import ReplaceOptim
+
+        self.optimizations = [
```

Comment:
We usually put these things on the class rather than the instance. Although maybe that's not such a great thing, because putting a mutable data structure on a class can be problematic if a subclass mutates it.